tendermint/tests/lite.rs

```diff
@@ -69,68 +61,45 @@ fn header_tests_verify() {
     run_test_cases(cases);
 }
 
-#[derive(Clone)]
-struct Trusted {
-    last_signed_header: SignedHeader,
-    validators: Set,
-}
-
-impl lite::TrustedState for Trusted {
-    type LastHeader = SignedHeader;
-    type ValidatorSet = Set;
-
-    fn new(last_header: &Self::LastHeader, vals: &Self::ValidatorSet) -> Self {
-        Self {
-            last_signed_header: last_header.clone(),
-            validators: vals.clone(),
-        }
-    }
-
-    fn last_header(&self) -> &Self::LastHeader {
-        &self.last_signed_header
-    }
-
-    fn validators(&self) -> &Self::ValidatorSet {
-        &self.validators
-    }
-}
+type Trusted = lite::TrustedState<SignedHeader, Header>;
 
 fn run_test_cases(cases: TestCases) {
     for (_, tc) in cases.test_cases.iter().enumerate() {
         let trusted_next_vals = tc.initial.clone().next_validator_set;
-        let mut trusted_state = Trusted::new(&tc.initial.signed_header.clone(), &trusted_next_vals);
+        let trusted_state =
+            Trusted::new(&tc.initial.signed_header.clone().into(), &trusted_next_vals);
         let expects_err = match &tc.expected_output {
             Some(eo) => eo.eq("error"),
             None => false,
         };
 
-        // TODO - we're currently using lite::verify_single which
-        // shouldn't even be exposed and doesn't check time.
-        // but the public functions take a store, which do check time,
-        // also take a store, so we need to mock one ...
-        /*
         let trusting_period: std::time::Duration = tc.initial.clone().trusting_period.into();
-        let now = tc.initial.now;
-        */
+        let tm_now = tc.initial.now;
+        let now = tm_now.to_system_time().unwrap();
+        // Note: Alternative to using mem-store from the light node crate would be mocking
+        // the store here directly.
+        let mut store = store::MemStore::new();
```

Comment:
We don't really need an externally defined store here, we can just use a cache again.